src/crypto/tls/handshake_client.go

```diff
@@ -131,19 +131,34 @@ func (c *Conn) makeClientHello(minVersion uint16) (*clientHelloMsg, ecdheParamet
 		}
 
 		curveID := config.curvePreferences()[0]
-		if _, ok := curveForCurveID(curveID); curveID != X25519 && !ok {
-			return nil, nil, errors.New("tls: CurvePreferences includes unsupported curve")
-		}
-		params, err = generateECDHEParameters(config.rand(), curveID)
-		if err != nil {
-			return nil, nil, err
+		if scheme := curveIdToCirclScheme(curveID); scheme != nil {
```

Comment:
Make sure this path is only taken if the client supports TLS 1.3.

Reply:
Yeah, probably. I'll add a check.

Reply:
This is within the `if hello.supportedVersions[0] == VersionTLS13` check.